src/hyperfine/export/mod.rs

```diff
@@ -0,0 +1,94 @@
+mod csv;
+
+use self::csv::CsvExporter;
+
+use std::io::Result;
+
+/// The ExportEntry is the main set of values that will
+/// be exported to files when requested.
+#[derive(Debug, Default, Clone, Serialize)]
+pub struct ExportEntry {
+    /// The command that was run
+    command: String,
+    /// The mean run time
+    mean: f64,
+    /// The standard deviation of all run times
+    stddev: f64,
+    /// Time spend in user space
+    user: f64,
+    /// Time spent in system space
+    system: f64,
+}
+
+impl ExportEntry {
+    /// Create a new ExportEntry with the given values
+    pub fn new(command: String, mean: f64, stddev: f64, user: f64, system: f64) -> Self {
+        ExportEntry {
+            command,
+            mean,
+            stddev,
+            user,
+            system,
+        }
+    }
+}
+
+/// The ResultExportType enum is used to denote the desired form
+/// of exporter to use for a given file.
+pub enum ResultExportType {
+    /// Export to a csv file with the provided name
+    Csv(String),
+}
+
+/// A ResultExporter is responsible for writing all results to the
+/// appropriate file
+trait ResultExporter {
+    /// Write all entries to the target destination
+    fn write(&self, values: &Vec<ExportEntry>) -> Result<()>;
+}
+
+/// The ExportManager handles coordination of multiple ResultExporters
+pub trait ExportManager {
```

Comment:
Do we need `ExportManager` to be a trait? Couldn't this just be a struct?

Reply:
Sure, I was just trying to extract away all implementation, but I can switch that one out

Reply:
I'd prefer a simple struct. I don't think we need `ExportManager` as an abstraction/interface here. There is just one of them. I think it would also simplify the code and the readability.